### PR TITLE
Add Client_ID to Audiences

### PR DESCRIPTION
**Pull request by jiaqi216**
GA4GH AAI requires that client ID be in `aud` in the access token if the `aud` claim is provided. Adding client id and also adding a test to check for compatibility with GA4GH spec. 

This might impact our existing validation if we're checking for audiences.

### CODE REVIEW

File: fence/jwt/token.py [hunk 1]
```diff
@@ -377,10 +377,14 @@ def generate_signed_access_token(
                 "must provide value for `iss` (issuer) field if"
                 " running outside of flask application"
             )
+    audiences = scopes.copy()
+    # include client_id in audiences as required by GA4GH
+    if client_id:
+        audiences.append(client_id)
```
Comment (vpsx) [inline fence/jwt/token.py:383]: The requirement on `aud` is actually not specific to GA4GH AAI but simply [a part of the generic JWT spec](https://tools.ietf.org/html/rfc7519#section-4.1.3). And the requirement is just that _if_ `aud` is supplied then it must include the `client_id`. So I would reword the comment to not include "as required by GA4GH". (I guess that would leave a very redundant comment so maybe just remove...)

Additionally, given the above, it might make more sense to start with `audiences = client_id if client_id else []`, and _then_ append the scopes to `audiences`, and add a comment explaining that we only append scopes for the sake of backwards compatibility. I think going forward that would just make a lot more sense to someone reading the code, versus starting from the scopes 🤪

Comment (jiaqi216) [reply fence/jwt/token.py:382]: Sounds good, made changes as you requested. Should be good to delete the `audiences = audiences + scope` part if/when we remove scopes from aud
